companies/serializers.py

```
@@ -1,14 +1,25 @@
+        return self.context.get('rank_dict', {}).get(obj.stock_code, None)
+    
+class CompanyRankingSerializer(serializers.ModelSerializer):
+    # stock_code(FK)을 통해 Company 모델의 필드에 접근합니다.
+    companyId = serializers.CharField(source='stock_code.stock_code')
+    name = serializers.CharField(source='stock_code.company_name')
+    logo = serializers.URLField(source='stock_code.logo_url')
+    amount = serializers.IntegerField(source='stock_code.market_amount')
+
+    class Meta:
+        model = CompanyRanking
+        fields = ['rank', 'companyId', 'name', 'logo', 'amount']
```

Ruff (0.14.10)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)